compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -2037,31 +2039,54 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
                   {
                   if (value >= -128 && value <= 127)
                      {
-                     opCode = IMULRegRegImms(nodeIs64Bit);
+                     opCode = IMulRegRegImms(node->getSize());
                      }
                   else
                      {
-                     opCode = IMULRegRegImm4(nodeIs64Bit);
+                     opCode = IMulRegRegImm4(node->getSize());
                      }
                   targetRegister = cg->evaluate(firstChild);
                   generateRegRegImmInstruction(opCode, node, targetRegister, targetRegister, value, cg);
                   }
                }
             }
          }
-      if (targetRegister)
+      }
+
+   //Evaluate the byte multiplier
+   if (!targetRegister  && dataType == TR::Int8)
+      {
+      targetRegister = cg->intClobberEvaluate(secondChild);
+      //Set register dependecy on EAX. AX = AL * r/m8
+      TR::RegisterDependencyConditions  *multDependencies = generateRegisterDependencyConditions((uint8_t)1, 1, cg);
+      multDependencies->addPreCondition(targetRegister, TR::RealRegister::eax, cg);
+      multDependencies->addPostCondition(targetRegister, TR::RealRegister::eax, cg);
+
+      if ((firstChild->getReferenceCount() == 1) && 
```

Comment:
nitpicking... the parentheses around firstChild->getOpCode().isMemoryReference() is redundant. I would suggest removing the parentheses around (firstChild->getReferenceCount() == 1) as well.

Reply:
Done